expo/gbe/views/review_act_view.py

```diff
@@ -19,111 +11,71 @@
-    '''
-    bid_eval = BidEvaluation.objects.filter(
-        bid_id=act_id,
-        evaluator_id=reviewer.resourceitem_id).first()
-    if bid_eval is None:
-        bid_eval = BidEvaluation(evaluator=reviewer, bid=act)
+    def groundwork(self, request, args, kwargs):
+        super(ReviewActView, self).groundwork(request, args, kwargs)
+        self.bidder = self.bidder_form_type(instance=self.object.performer,
+                                            prefix=self.bidder_prefix)
+
+        audio_info = self.object.tech.audio
+        stage_info = self.object.tech.stage
+        initial = {
+            'track_title': audio_info.track_title,
+            'track_artist': audio_info.track_artist,
+            'track_duration': audio_info.track_duration,
+            'act_duration': stage_info.act_duration
+        }
+        self.create_object_form(initial=initial)
+        self.readonlyform_pieces = [self.object_form, self.bidder]
+
+    def get(self, request, *args, **kwargs):
+        self.groundwork(request, args, kwargs)
+        self.form = BidEvaluationForm(instance=self.bid_eval)
+        return (self.object_not_current_redirect() or
+                self.bid_review_response(request))
 
-    # show act info and inputs for review
-    if request.method == 'POST':
-        form = BidEvaluationForm(request.POST, instance=bid_eval)
-        if form.is_valid():
-            evaluation = form.save(commit=False)
-            evaluation.evaluator = reviewer
-            evaluation.bid = act
-            evaluation.save()
-            return HttpResponseRedirect(reverse('act_review_list',
-                                                urlconf='gbe.urls'))
-        else:
-            return render(request, 'gbe/bid_review.tmpl',
-                          {'readonlyform': [actform, performer],
-                           'reviewer': reviewer,
-                           'form': form,
-                           'actionform': actionform,
-                           'actionURL': actionURL,
-                           'conference': conference,
-                           'old_bid': old_bid,
-                           })
-    else:
-        form = BidEvaluationForm(instance=bid_eval)
-        return render(request,
-                      'gbe/bid_review.tmpl',
-                      {'readonlyform': [actform, performer],
-                       'reviewer': reviewer,
-                       'form': form,
-                       'actionform': actionform,
-                       'actionURL': actionURL,
-                       'conference': conference,
-                       'old_bid': old_bid,
-                       })
+    def post(self, request, *args, **kwargs):
+        self.groundwork(request, args, kwargs)
+        self.form = BidEvaluationForm(request.POST, instance=self.bid_eval)
+        return (self.object_not_current_redirect() or
+                self.post_response_for_form(request))
```

Comment:
We need both functions because the dispatch method is going to call one or the other depending on the request method. Could try to abstract out a little more, but it seems like we've hit a point of diminishing returns.